crypto/dsa/dsa_ossl.c

```diff
@@ -189,17 +193,14 @@ static int dsa_sign_setup(DSA *dsa, BN_CTX *ctx_in,
 
     /*
      * We do not want timing information to leak the length of k, so we
-     * compute g^k using an equivalent exponent of fixed length. (This
-     * is a kludge that we need because the BN_mod_exp_mont() does not
-     * let us specify the desired timing behaviour.)
+     * compute ensure the k is longer than the q value.  Any code that checks
+     * for this and does a modulo redution will consequently also be consant
+     * time.  This is a kludge that we need because the BN_mod_exp_mont()
```

Comment:
The description kind of gets a bit tricky here. If BN_mod_exp_mont did the reduction, then it would effectively give away length. Does it make sense to clarify that "Any code [other than modexp]"?

Reply:
Changed.